packages/google_maps_flutter/google_maps_flutter_android/pigeons/messages.dart

```diff
@@ -156,26 +156,61 @@ class PlatformMarker {
 
 /// Pigeon equivalent of the Polygon class.
 class PlatformPolygon {
-  PlatformPolygon(this.json);
+  PlatformPolygon({
+    required this.polygonId,
+    required this.consumesTapEvents,
+    required this.fillColor,
+    required this.geodesic,
+    required this.points,
+    required this.holes,
+    required this.visible,
+    required this.strokeColor,
+    required this.strokeWidth,
+    required this.zIndex,
+  });
 
-  /// The polygon data, as JSON. This should only be set from
-  /// Polygon.toJson, and the native code must interpret it according to the
-  /// internal implementation details of that method.
-  // TODO(stuartmorgan): Replace this with structured data. This exists only to
-  //  allow incremental migration to Pigeon.
-  final Map<String?, Object?> json;
+  final String polygonId;
+  final bool consumesTapEvents;
+  final int fillColor;
+  final bool geodesic;
+  final List<PlatformLatLng?> points;
+  final List<List<PlatformLatLng?>?> holes;
+  final bool visible;
+  final int strokeColor;
+  final int strokeWidth;
+  final int zIndex;
 }
 
 /// Pigeon equivalent of the Polyline class.
 class PlatformPolyline {
-  PlatformPolyline(this.json);
+  PlatformPolyline({
+    required this.polylineId,
+    required this.consumesTapEvents,
+    required this.color,
+    required this.geodesic,
+    required this.jointType,
+    required this.patterns,
+    required this.points,
+    required this.startCap,
+    required this.endCap,
+    required this.visible,
+    required this.width,
+    required this.zIndex,
+  });
 
-  /// The polyline data, as JSON. This should only be set from
-  /// Polyline.toJson, and the native code must interpret it according to the
-  /// internal implementation details of that method.
-  // TODO(stuartmorgan): Replace this with structured data. This exists only to
-  //  allow incremental migration to Pigeon.
-  final Map<String?, Object?> json;
+  final String polylineId;
+  final bool consumesTapEvents;
+  final int color;
+  final bool geodesic;
+  final int jointType;
+  // TODO(schectman): convert to structured data.
+  final List<Object?> patterns;
+  final List<PlatformLatLng?> points;
+  final Object startCap;
+  final Object endCap;
```

Comment:
`startCap` and `endCap` should also have TODOs to make them structured, and like the old `json` TODO it should clearly indicate what the exact code populating the JSON.